architecture/rbac.md

```diff
@@ -0,0 +1,111 @@
+# RBAC
+
+
+## How it works
+
+RBAC in ManageIQ is split into 2 parts; what you can "see" and what you can "do".
+
+**What you can "see"** is defined as access to resources based on a number of criteria and rules. (Tenancy, Ownership, Entitlements...)
+
+**What you can "do"** is defined by permissions to perform actions. Such permission could be tied to certain kinds of resources. For example - it allows you to specify permission for a user to  edit or delete (actions) Virtual Machines(kind of resource). These are called Product Features in ManageIQ.
+
+At the heart of the RBAC system are Users and Groups. A user can belong to multiple groups, and while a user is interacting with ManageIQ there is the notion of a "current" group. Only the current group's permission set will apply during that session, or until the "current" group is changed. Both a user and a group can directly "own" resources in the system.
+
+A Group belongs to a Tenant. Tenants are a hierarchical structure above groups with inheritance rules for the resources they "own".
+
+A Group is mapped to a Role, and a Role is a collection of a number of Product Features and Entitlements. Product Features are permissions for actions which you can "do" with the system. Entitlements are a set of ways of grouping resources such that a user can "see" them.(Entitlements is just one part from criterias defined by  what you can "see" aspect)
+
+![](images/rbac.svg)
+
+
+## What you can "see"
+There are several possible ways how resources could be accessed by users and there are combinations of those ways.
+
+- **Ownership**
+
+A resource in ManageIQs can be directly owned by a User or Group. A resource can additionally be owned by a tenant. All objects directly owned by a user or owned by the user's current group can be seen.
+
+
+- **Tenancy**
+
+Tenancy ownership in ManageIQs is determined by first a direct ownership to a resource, and then applying certain "ancestor" or "descendant" rules depending on the type of resource being viewed.
+
+
+![Tenancy](images/tenancy.svg)
+
+For example, a member of a tenant can only see virtual templates owned by their tenant or owned by any parent tenants. This "ancestor" view is used to allow for sharing a virtual template across multiple child tenants. In contrast, a member of a tenant can only see virtual machine instances owned by their tenant or owned by any child tenants. This "descendant" view is used to allow the parent tenant to do accounting of all running virtual machines instances below them.
```

Comment:
For example, a member of a tenant can only see virtual templates owned by their tenant or owned by any parent tenants. This "ancestor" view is used to allow for sharing a virtual template ** across multiple child tenants**....

Shouldn't be here **across multiple parent tenants **?